spec/tapioca/compilers/dsl/active_record_relations_spec.rb

```diff
@@ -0,0 +1,602 @@
+            def forty_two; end
+
+            sig { returns(Post) }
+            def forty_two!; end
+
+            sig { returns(T.nilable(Post)) }
+            def fourth; end
+
+            sig { returns(Post) }
+            def fourth!; end
+
+            sig { returns(Array) }
+            def ids; end
+
+            sig { params(limit: T.untyped).returns(T.untyped) }
+            def last(limit = nil); end
+
+            sig { returns(Post) }
+            def last!; end
+
+            sig { params(block: T.nilable(T.proc.params(record: Post).returns(T.untyped))).returns(T::Boolean) }
+            def many?(&block); end
+
+            sig { params(column_name: T.any(String, Symbol)).returns(T.untyped) }
+            def maximum(column_name); end
+
+            sig { params(column_name: T.any(String, Symbol)).returns(T.untyped) }
+            def minimum(column_name); end
+
+            sig { params(attributes: T.nilable(T.any(::Hash, T::Array[::Hash])), block: T.nilable(T.proc.params(object: Post).void)).returns(Post) }
+            def new(attributes = nil, &block); end
+
+            sig { params(block: T.nilable(T.proc.params(record: Post).returns(T.untyped))).returns(T::Boolean) }
+            def none?(&block); end
+
+            sig { params(block: T.nilable(T.proc.params(record: Post).returns(T.untyped))).returns(T::Boolean) }
+            def one?(&block); end
+
+            sig { params(column_names: T.untyped).returns(T.untyped) }
+            def pluck(*column_names); end
+
+            sig { returns(T.nilable(Post)) }
+            def second; end
+
+            sig { returns(Post) }
+            def second!; end
+
+            sig { returns(T.nilable(Post)) }
+            def second_to_last; end
+
+            sig { returns(Post) }
+            def second_to_last!; end
+
+            sig { params(column_name: T.nilable(T.any(String, Symbol)), block: T.nilable(T.proc.params(record: T.untyped).returns(T.untyped))).returns(T.untyped) }
+            def sum(column_name = nil, &block); end
+
+            sig { params(limit: T.untyped).returns(T.untyped) }
+            def take(limit = nil); end
+
+            sig { returns(Post) }
+            def take!; end
+
+            sig { returns(T.nilable(Post)) }
+            def third; end
+
+            sig { returns(Post) }
+            def third!; end
+
+            sig { returns(T.nilable(Post)) }
+            def third_to_last; end
+
+            sig { returns(Post) }
+            def third_to_last!; end
+          end
+
+          module GeneratedAssociationRelationMethods
+            sig { returns(PrivateAssociationRelation) }
+            def all; end
+
+            sig { params(args: T.untyped, blk: T.untyped).returns(PrivateAssociationRelation) }
+            def create_with(*args, &blk); end
+
+            sig { params(args: T.untyped, blk: T.untyped).returns(PrivateAssociationRelation) }
+            def create_with!(*args, &blk); end
```

Comment:
Those methods with `!` should not be generated. They are private API of the framework and applications should not be calling.

Reply:
Done. Except it seems `uniq!` is public API, so I included that, but it doesn't appear here since this tests against Rails 5.2